reactioncommerce/federated-gateway
# feat: add gateway level authorization service check

Resolves #13 
Impact: **minor**
Type: **feature**

## Summary
Adds a Gateway level check to make sure the user trying to make a request through the gateway is allowed to use the gateway.

Currently, the policy - added at `bin/policies` - states all users are allowed to use the gateway, so this won't do anything aside from prove the check works.

Since the check requires the `reaction-authorization` service to be running, this authorization check is behind a `ENABLE_AUTHORIZTION` flag, defaulted to `false`, in the `env`.

## Testing
1. See that authorization service is not hit if `ENABLE_AUTHORIZTION` flag is `false`
1. See that authorization service is hit if `ENABLE_AUTHORIZTION` flag is `true`
1. in `index.js`, hardcode `isAuthorized` to return `false`
1. See that this throws an auth error when trying to make any query or mutation - even those that typically were "unauthenticated" in the `api`
1. See that all users are allowed to make all requests if you let the authorization service run the check
